### DISCUSSION TITLE
How to add new entity types

### DISCUSSION THREAD

**Discussion by fbruzzaniti** [alephdata/aleph#2761] [Q&A] [answered]
I'd like to add some new entity types, could you give me a rough idea of what files I'd need to alter, the entities I was thinking of adding were:

Various crypto addresses
Webmoney
domains
(IP's but I think someone is already adding those by the looks of it)
Australian bank account numbers (by looking for the BSB/Branch numbers)
Hashes

Or is their a way to do it with the API. I.e. used the API to read the data then return any entities so you can interpolate Aleph with 3rd party library's or utilities without need to change the source.

**Comment by tillprochaska** [comment, marked as answer]
The entity types (schemata) Aleph uses are defined by the FollowTheMoney (FtM) data model: https://followthemoney.readthedocs.io.

If you think a new schema should be added to FtM, I’d recommend to discuss this in the Slack community or to open an issue in the FtM repository: https://github.com/alephdata/followthemoney.

However, there may already be schemata in FtM that could suit your use case, for example:

* [Crpytocurrency wallet](https://followthemoney.readthedocs.io/en/latest/model.html#cryptocurrency-wallet)
* [Web page](https://followthemoney.readthedocs.io/en/latest/model.html#web-page)
* [Asset](https://followthemoney.readthedocs.io/en/latest/model.html#asset)
* [Bank account](https://followthemoney.readthedocs.io/en/latest/model.html#bank-account)
